automation/jenkins-hostapp-extension.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash 
+
+set -e
+
+[ $# -lt 2 ] && echo "Hostapp extension packages need to be defined" && exit 1
+[ -z "${BALENAOS_STAGING_TOKEN}" ] && echo "Balena staging token is required" && exit 1
+[ -z "${BALENAOS_PRODUCTION_TOKEN}" ] && echo "Balena production token is required" && exit 1
+[ -z "${HOSTEXT_NAME}" ] && echo "Hostapp extension name is required" && exit 1
+[ -z "${deployTo}" ] && echo "Deployment environment target is required" && exit 1
+[ -z "${VERSION_HOSTOS}" ] && echo "Balena OS version is required" && exit 1
+[ -z "${MACHINE}" ] && echo "Device type is required" && exit 1
+[ -z "${PACKAGE_TYPE}" ] && echo "Package type is required" && exit 1
+
+PACKAGES=""
+for pkg in $@; do
+	if [ -z "${PACKAGES}" ]; then
+		PACKAGES="${pkg}"
+	else
+		PACKAGES="${PACKAGES} ${pkg}"
+	fi
+done
+
+finish() {
+	result=$?
+	exit ${result}
+}
+trap finish EXIT ERR
+
+deploy_to_balena() {
+	local _exported_image_path=$1
+	local DEVELOPMENT_IMAGE=no
+	if [ "${buildFlavor}" = "dev" ]; then
+		DEVELOPMENT_IMAGE=yes
+	fi
+	docker pull ${NAMESPACE}/balena-push-env
+	docker run --rm -t \
+		-e BASE_DIR=/host \
+		-e BALENAOS_STAGING_TOKEN="${BALENAOS_STAGING_TOKEN}" \
+		-e BALENAOS_PRODUCTION_TOKEN="${BALENAOS_PRODUCTION_TOKEN}" \
+		-e APPNAME="${HOSTEXT_NAME}" \
+		-e DEVELOPMENT_IMAGE="${DEVELOPMENT_IMAGE}" \
+		-e DEPLOY_TO="${deployTo}" \
+		-e VERSION_HOSTOS="${VERSION_HOSTOS}" \
+		-e ESR="${ESR}" \
+		-e META_BALENA_VERSION="${META_BALENA_VERSION}" \
+		-v "${_exported_image_path}":/host/appimage.docker \
+		--privileged \
+		${NAMESPACE}/balena-push-env /balena-push-os-version.sh
+}
+
+WORKSPACE=${WORKSPACE:-"${PWD}"}
+
+DEVICE_TYPE_JSON="$WORKSPACE/$MACHINE.json"
+if [ -e "${DEVICE_TYPE_JSON}" ]; then
+    SLUG=$(jq --raw-output '.slug' "$DEVICE_TYPE_JSON")
+fi
+[ -z "${SLUG}" ] && echo "Slug is required" && exit 1
```

Comment:
Thew references to SLUG above can be removed since it does not look like they are used for host extensions

Reply:
@floion you're right. When coding I was unsure as to go with MACHINE or SLUG, and I went for MACHINE.